iceoryx_binding_c/include/iceoryx_binding_c/posh_runtime.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_BINDING_C_POSH_RUNTIME_H
+#define IOX_BINDING_C_POSH_RUNTIME_H
+
+#include "internal/c2cpp_binding.h"
+
+/// @brief registers a new posh runtime with a given name
+/// @param[in] name name of the posh runtime
+void iox_rt_getInstance(const char* const name);
```

Comment:
it's a bit weird to not have a return value when the function has `get` in the name. I know that there is a singleton under the hood, but maybe we can either return the handle, which makes it somehow unergonomic because you don't really need that handle or we use a different name.
I know you like it to be as close as possible to the C++ API, but in this case I would make an exception.
Also, the mixed snake_case and CamelCase doesn't look good. What do you think by using `method_name` instead of `methodName`?
The last thing to think about is the abbreviation `_rt_` for `runtime`. Unfortunately this has several meanings and could also be interpreted as `realtime`. I don't have that strong feelings about this, but `_rte_` for `runtime environment` might be an option or just `_runtime_`. These functions aren't used that often, so it might be a good trade-off between the amount of typing and the unambiguity of the function name

Reply:
fixed by renaming the function to `iox_runtime_register`. Everything is now in  snake_case